ui-v2/app/adapters/application.js

```diff
@@ -21,13 +22,44 @@ export default Adapter.extend({
       ...this._super(...arguments),
     };
   },
+  handleBooleanResponse: function(url, response, primary, slug) {
+    return {
+      // consider a check for a boolean, also for future me,
+      // response[slug] // this will forever be null, response should be boolean
+      [primary]: this.uidForURL(url /* response[slug]*/),
+    };
+  },
+  // could always consider an extra 'dc' arg on the end here?
+  handleSingleResponse: function(url, response, primary, slug, _dc) {
+    const dc =
+      typeof _dc !== 'undefined' ? _dc : url.searchParams.get(DATACENTER_QUERY_PARAM) || '';
+    return {
+      ...response,
+      ...{
+        [DATACENTER_KEY]: dc,
+        [primary]: this.uidForURL(url, response[slug]),
+      },
+    };
+  },
+  handleBatchResponse: function(url, response, primary, slug) {
+    const dc = url.searchParams.get(DATACENTER_QUERY_PARAM) || '';
+    return response.map((item, i, arr) => {
+      return this.handleSingleResponse(url, item, primary, slug, dc);
+    });
+  },
   cleanQuery: function(_query) {
     delete _query.id;
     const query = { ..._query };
-    delete _query[DATACENTER_KEY];
+    delete _query[DATACENTER_QUERY_PARAM];
     return query;
   },
-  isQueryRecord: function(url) {
+  isUpdateRecord: function(url, method) {
+    return false;
+  },
+  isCreateRecord: function(url, method) {
+    return false;
+  },
```

Comment:
What's happening here with the always return false? Are these the defaults that are overridden by individual adapters?

Reply:
Yes exactly